corehq/apps/data_interfaces/deduplication.py

```diff
@@ -171,10 +171,8 @@ def backfill_deduplicate_rule(domain, rule):
         )
     finally:
         progress_helper.set_rule_complete()
-        AutomaticUpdateRule.objects.filter(pk=rule.pk).update(
-            locked_for_editing=False,
```

Comment:
doesn't this also need to be updated?

Reply:
Thanks Simon. I only checked for `AutomaticUpdateRule.by_model` 🤦🏻. 

Also curious to hear if you have strong opinions on `AutomaticUpdateRule.objects.filter(rule=rule.pk).update(...)` vs `rule.save(update_field=[...])`. I figured in the cases where we already had access to the `rule` db object, using the latter method made sense, but open to changes.

Reply:
This still isn't setting `locked_for_editing` to False like it was before

Reply:
See 1e2cc960a39e3513473e39d055929486f779a320